OffscreenCanvas WebGL tests in webgl/2.0.y/conformance/textures/webgl_canvas fail.

mattwoodrow opened the pull request:
#### 3894e0d4567012240c6cf8a29d13127531d94770
<pre>
OffscreenCanvas WebGL tests in webgl/2.0.y/conformance/textures/webgl_canvas fail.
<a href="https://bugs.webkit.org/show_bug.cgi?id=256479">https://bugs.webkit.org/show_bug.cgi?id=256479</a>
&lt;rdar://problem/109051251&gt;

Reviewed by Dan Glastonbury.

Resizing the OffscreenCanvas object wasn&apos;t communicating this through to WebGL, which needs
to know about the new size for its framebuffer.
This code matches what HTMLCanvasElement does when resized.

* LayoutTests/TestExpectations:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::OffscreenCanvas::setSize):

Canonical link: <a href="https://commits.webkit.org/263876@main">https://commits.webkit.org/263876@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec9571148e50aef4d322a80a2eddd94d29386233

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5972 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6148 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6333 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7526 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6330 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5970 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6367 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6107 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/8905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6080 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6117 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5410 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7586 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3594 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5386 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13317 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5460 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/5465 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/7697 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5918 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4857 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5350 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1418 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9477 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5712 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebCore/html/OffscreenCanvas.cpp:
```diff
@@ -180,8 +180,12 @@ void OffscreenCanvas::setHeight(unsigned newHeight)
     reset();
+
+    if (m_context && m_context->isGPUBased() && oldSize != size())
+        downcast<GPUBasedCanvasRenderingContext>(*m_context).reshape(width(), height());
```
`reshape` checks for `width` and `height` being different to it's current value, so you could skip the `oldSize != size()`

djg reviewed (approved):
:shipit:

mattwoodrow committed:
OffscreenCanvas WebGL tests in webgl/2.0.y/conformance/textures/webgl_canvas fail.
https://bugs.webkit.org/show_bug.cgi?id=256479
<rdar://problem/109051251>

Reviewed by Dan Glastonbury.

Resizing the OffscreenCanvas object wasn't communicating this through to WebGL, which needs
to know about the new size for its framebuffer.
This code matches what HTMLCanvasElement does when resized.

* LayoutTests/TestExpectations:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::OffscreenCanvas::setSize):

Canonical link: https://commits.webkit.org/263876@main
